### ISSUE TITLE
Creating flow outputs of type ARRAY fails and doesn't generate an error log 

### ISSUE THREAD

**Issue by Ben8t** [kestra-io/kestra#4769]
### Describe the issue

This flow works:

```yaml
id: array_output
namespace: kestra.sandbox
description: Save and Execute the flow

tasks:
  - id: return
    type: io.kestra.plugin.core.output.OutputValues
    values:
      one: "1rstValue"
      two: "2ndValue"

outputs:
  - id: myout
    type: ARRAY
    value: '["{{ outputs.return.values.one }}", "{{ outputs.return.values.two }}"]'
```

but this one with "dash" syntax doesn't work (also tried with several variations of quotes)

```yaml
id: array_output
namespace: kestra.sandbox
description: Save and Execute the flow

tasks:
  - id: return
    type: io.kestra.plugin.core.output.OutputValues
    values:
      one: "1rstValue"
      two: "2ndValue"

outputs:
  - id: myout
    type: ARRAY
    value:
      - '"{{ outputs.return.values.one }}"'
      - '"{{ outputs.return.values.two }}"'
```

raising the following error:

ERROR jdbc-queue-Execution_0 flow.array_output [namespace: kestra.sandbox] [flow: array_output] [execution: 19JVhLwglJ0nprYzdkVrKq] Failed to render output values
jakarta.validation.ConstraintViolationException: myout: Invalid output for `myout`, Expected `ARRAY` but received `["1rstValue", "2ndValue"]` with errors:
```
class java.util.ArrayList cannot be cast to class java.lang.String (java.util.ArrayList and java.lang.String are in module java.base of loader 'bootstrap')
```, but received `["1rstValue", "2ndValue"]`

### Environment

- Kestra Version: 0.18.3
- Operating System (OS/Docker/Kubernetes): docker
- Java Version (if you don't run kestra in Docker):

**Comment by anna-geller** [comment]
Nice, you spotted even two bugs! 😄 

First bug — this should work:

```yaml
id: array_output
namespace: kestra.sandbox

tasks:
  - id: return
    type: io.kestra.plugin.core.output.OutputValues
    values:
      one: "1rstValue"
      two: "2ndValue"

outputs:
  - id: myout
    type: ARRAY
    value: 
      - "{{ outputs.return.values.one }}"
      - "{{ outputs.return.values.two }}"
# this should also work:
    value: ["{{ outputs.return.values.one }}", "{{ outputs.return.values.two }}"]
```

Second bug — that execution will currently fail but it doesn't generate any log about why. We should log an error that can be displayed in the Temporal view if outputs can't be parsed:

![image](https://github.com/user-attachments/assets/0d6b69f0-adfe-406d-a3e7-8996b99f80df)

**Comment by loicmathieu** [comment]
~@Ben8t I just tested it with our develop version and it works, it seems to have already been fixed in 0.19.0-SNAPSHOT.
Can you double-check and confirm it is now fixed?~

My bad, I can reproduce it.
